### PR TITLE
feat(app): show provider config details on hover in eval results

### PR DESCRIPTION
**Pull request by mldangelo**
## Summary

- Add hover tooltip to provider names in eval results table showing full configuration
- Sanitize configs to redact secrets while preserving field names for debugging context
- Match providers by ID/label instead of fragile index-based matching
- Enhance backend sanitizer with value-based secret detection (sk-..., Bearer tokens, AWS keys, etc.)

## Details

**Problem:** The eval results table only showed the provider string (e.g., "openai:gpt-4o") with no visibility into the actual configuration used (temperature, reasoning_effort, thinking config, etc.).

**Solution:** New `ProviderDisplay` component shows provider name normally, but reveals full sanitized config on hover:
- YAML-formatted config in tooltip
- Shows provider ID when custom label differs
- Redacts sensitive values (`[REDACTED]`) but keeps field names visible
- Excludes non-useful fields (callApi, transform, env)

**Backend sanitizer improvements:**
- New `looksLikeSecret()` detects secrets by value patterns (OpenAI keys, Anthropic keys, Bearer tokens, AWS keys, Google API keys, long base64 strings)
- Exports utilities for frontend reuse (DRY)
- Adds missing sensitive fields: credentials, apisecret, apikeyenvar

## Test plan

- [x] All 2217 frontend tests pass
- [x] All 163 backend sanitizer tests pass
- [x] 58 new tests for ProviderDisplay and providerConfig modules
- [ ] Manual testing: hover over provider names in eval results to see config tooltip

### CODE REVIEW

**Comment by will-holley** [comment]
I'm seeing this when a provider has no config:

image

Also id duplicatation:

image

**Review by will-holley — CHANGES_REQUESTED** [review]
See prev. comment.

**Comment by faizanminhas** [comment]
LGTM:
Screenshot 2026-01-02 at 12 44 59 PM

Also confirmed that in the no-config case, there's just simply no hover/preview:
Screenshot 2026-01-02 at 12 45 31 PM

**Review by faizanminhas — APPROVED** [review]
brilliant change, very inspired

File: src/app/src/pages/eval/components/ProviderDisplay.tsx [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+/**
+ * Recursively sanitize config for display, redacting sensitive values.
+ * Shows field names but hides secret values.
+ *
+ * This extends the shared sanitizer with frontend-specific features:
+ * - Excluded fields (callApi, transform, env, etc.)
+ * - Returns undefined for empty objects (cleaner tooltip display)
+ * - Higher depth limit (10 vs 4) for detailed config display
+ */
+export function sanitizeConfig(obj: any, depth = 0, parentKey = ''): any {
```
Comment (faizanminhas) [inline src/app/src/pages/eval/components/ProviderDisplay.tsx]: what's the point of redacting? if the user clicks "view yaml" in eval actions, don't they also see the secrets in the config?

**Comment by ghost** [comment]
## ✅ Generated 79 tests - 79 passed (c6f2a0a8777f5fae66b499545abb97f1d8e6de92) [View tests ↗](https://app.usetusk.ai/app/testing-commit-check/678f7bba-2309-4b87-8012-239aec408534?client=e91e3433-8fb4-4e98-b233-bba492f4909c)

### Test Summary

- `ProviderDisplay` - 9 ✅
- `filterConfigForDisplay` - 12 ✅
- `extractProviderString` - 34 ✅
- `extractConfigBadges` - 13 ✅
- `findProviderConfig` - 11 ✅

### Results

Tusk's tests are all passing and demonstrate solid coverage of the new provider configuration display feature. The test suite validates three critical areas: (1) `filterConfigForDisplay` correctly sanitizes and structures provider configs for UI display, handling edge cases like nested objects, recursion limits, and falsy values; (2) `ProviderDisplay` component renders configs accurately in tooltips with proper formatting for various data types; and (3) `findProviderConfig` reliably matches providers by ID/label instead of fragile index-based matching, with proper fallback logic. The `extractProviderString` tests are particularly thorough, covering 34 scenarios including edge cases like circular references, non-standard input types, and null/undefined handling. `extractConfigBadges` validates that configuration parameters are correctly extracted and formatted for display (e.g., token counts with 'k'/'M' suffixes, temperature/penalty defaults). Together, these tests protect the core logic that enables users to inspect full provider configurations on hover in eval results—a key usability improvement for debugging model behavior.

---
<details><summary>View check history</summary>
<p>

| Commit | Status | Output | Created (UTC) |
|--------|--------|--------|--------|
| 4a4e7970f9b18a6862024372beb69d8aefb296d9 | ⏩ No tests generated | [Output](https://app.usetusk.ai/app/testing-commit-check/29760353-7f7a-46eb-bf5f-cc1a33aa9ade?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Dec 18, 2025 5:50AM |
| fc5789bc61e868767c2376b0761174dbae11f14a | ⏩ No tests generated | [Output](https://app.usetusk.ai/app/testing-commit-check/7a260a5f-6a39-4ca2-a3bf-a4fc901dfcf2?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Dec 18, 2025 6:41AM |
| 0c1a1e4cfbcfc30617720306b4c1d857d99266f0 | ⏩ No tests generated | [Output](https://app.usetusk.ai/app/testing-commit-check/d48d1dba-6a3c-43b2-9e89-4a046171d636?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Dec 18, 2025 7:02AM |
| 4986a865101be3d6bc5884984e15301799c3dc64 | 🔄 Running Tusk Tester | [Output](https://app.usetusk.ai/app/testing-commit-check/54981654-fd2e-4928-ac86-1781e11c5eb8?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Dec 20, 2025 7:04AM |
| fb2284d495a3ffd14a0d9976174d9a1ebcabde35 | ⏩ Skipped due to new commit on branch | [Output](https://app.usetusk.ai/app/testing-commit-check/59e0ea8a-5206-4a93-bbf4-878a51cc52e3?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Dec 28, 2025 8:01AM |
| aa901513fdf436a9f4d16ae28699b53a51734d6e | 🔄 Running Tusk Tester | [Output](https://app.usetusk.ai/app/testing-commit-check/f763c5f5-6420-46d2-9107-ae77dff38caf?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Dec 28, 2025 8:06AM |
| d0eb987aac185868fe7277e6c4fd35bfa91beb6a | ⏩ No test scenarios generated | [Output](https://app.usetusk.ai/app/testing-commit-check/8a04e422-b524-4903-b10d-06719084674d?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Jan 2, 2026 5:10AM |
| 62b45bf16ece3ec4d3160ab88be28be16b97df18 | ⏩ No test scenarios generated | [Output](https://app.usetusk.ai/app/testing-commit-check/b2dbd21e-7304-4566-a16a-366bb7f3157a?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Jan 10, 2026 11:52PM |
| 8595214fcfbf43cf1d9c1fb80daecb3745fe487d | ⏩ No test scenarios generated | [Output](https://app.usetusk.ai/app/testing-commit-check/5086bdce-a80b-4ef6-9b80-08abb0916d33?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Jan 10, 2026 11:57PM |
| 0f69a943c41af314d5f558e6cccf955304a337d1 | ✅ Generated 73 tests - 73 passed | [Tests](https://app.usetusk.ai/app/testing-commit-check/d63c90f5-0e50-440e-9bbd-18af8d4b68ff?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Jan 11, 2026 12:21AM |
| 38f8421ae93332772eec17bd424cb32ab038f1f3 | ⏩ Skipped due to new commit on branch | [Output](https://app.usetusk.ai/app/testing-commit-check/a0eed118-2ebb-467a-be2e-830a3c735143?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Jan 11, 2026 12:57AM |
| c6f2a0a8777f5fae66b499545abb97f1d8e6de92 | ✅ Generated 79 tests - 79 passed | [Tests](https://app.usetusk.ai/app/testing-commit-check/678f7bba-2309-4b87-8012-239aec408534?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Jan 11, 2026 1:07AM |

</p>
</details>

[View output in GitHub ↗](https://github.com/promptfoo/promptfoo/pull/6757/checks?check_run_id=60012247876)

_Was Tusk helpful? Give feedback by reacting with 👍 or 👎_
